src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -95,5 +97,33 @@ public static class IOExtensions {
             }
             return Path.DirectorySeparatorChar.ToString();
         }
+
+        public static bool HasWritePermission(this string path) {
```

Comment:
I don't think that is the case for Windows ACLs, at least - I just tried, and you can deny read and allow write to the same user at the same time, and it actually enforces it in that manner.